src/Axiom/Set/TotalMapOn.agda

```diff
@@ -0,0 +1,65 @@
+{-# OPTIONS --safe --no-import-sorts #-}
+
+open import Axiom.Set using (Theory)
+
+module Axiom.Set.TotalMapOn (th : Theory) where
+
+open import Prelude hiding (lookup)
+
+open import Agda.Primitive              using () renaming (Set to Type)
+open import Axiom.Set.Map th            using (left-unique ; Map ; mapWithKey-uniq)
+open import Axiom.Set.Rel th            using (Rel ; dom ; dom∈)
+open import Interface.DecEq             using (DecEq ; _≟_)
+open import Relation.Nullary.Decidable  using (Dec ; yes ; no)
+
+open Theory th    using ( Set ; _⊆_ ; _∈_ ; map ; ∈-map′ )
+open Equivalence  using (to)
+
+private variable A B : Type
+
+_TotalOn_ : Rel A B → Set A → Type
+R TotalOn X = X ⊆ dom R
+
+
+record TotalMapOn {A : Type}(X : Set A)(B : Type) : Type where
+  field  rel              : Set (A × B)
+         left-unique-rel  : left-unique rel
+         total-rel        : rel TotalOn X
+
+  domain : Set A
```

Comment:
Now that I look at this, I think I'd remove that. It's not actually the domain, since we just have `domain ⊆ dom rel`, but not the other inclusion, and it's confusing to have the two that aren't actually equal.

Reply:
I agree. I'll remove it and then merge.

Thanks for all the helpful comments!